### PR TITLE
Add CI/CD maintenance docs, K8s version details

### PR DESCRIPTION
**Pull request by dfarrell07**
Add website subsection dedicated to the maintenance of Submariner's
CI/CD systems. Document the K8s versions tested in Submariner and how
they should be maintained.

Relates-to: #624
Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

### CODE REVIEW

File: src/content/development/building-testing/ci-maintenance/_index.en.md
Comment (nyechiel) [inline src/content/development/building-testing/ci-maintenance/_index.en.md]: I think it would be useful to link to the main Kubernetes releases page which is the source of truth for support dates, EOL, etc: https://kubernetes.io/releases/

File: src/content/development/building-testing/ci-maintenance/_index.en.md
Comment (nyechiel) [inline src/content/development/building-testing/ci-maintenance/_index.en.md:2]: Since you are adding a new page, don't forget to update the index page as well: https://submariner.io/development/ (can also be addressed in a separate PR)

File: src/content/development/building-testing/ci-maintenance/_index.en.md
Comment (nyechiel) [inline src/content/development/building-testing/ci-maintenance/_index.en.md]:
```suggestion
This page documents the maintenance of Submariner's CI/CD for developers.
```

File: src/content/development/building-testing/ci-maintenance/_index.en.md
Comment (nyechiel) [inline src/content/development/building-testing/ci-maintenance/_index.en.md]: Any reason not to expand "K8s" to “Kubernetes” throughout this doc? https://submariner.io/development/website/style-guide/

**Comment by dfarrell07** [comment]
Thanks for the review @nyechiel, issues resolved!

File: src/content/development/building-testing/ci-maintenance/_index.en.md
Comment (dfarrell07) [reply src/content/development/building-testing/ci-maintenance/_index.en.md]: I had a dev audience in mind, but yes you're right better to be consistent.

**Review by nyechiel — APPROVED** [review]

File: src/content/development/building-testing/ci-maintenance/_index.en.md
Comment (mkolesnik) [inline src/content/development/building-testing/ci-maintenance/_index.en.md:23]: This sounds more like a job (or even unit test) that needs to happen on `subctl` only, doesn't make sense otherwise.

**Review by mkolesnik — DISMISSED** [review]

File: src/content/development/building-testing/ci-maintenance/_index.en.md
Comment (dfarrell07) [reply src/content/development/building-testing/ci-maintenance/_index.en.md:23]: A unit test for `subctl` could only test that we are acting on the cut off defined by `minK8sMinor`, it can't tell us if Submariner and all its features actually work with 1.17. That's why we agreed we need an E2E test for the cut-off version.

**Comment by dfarrell07** [comment]
@mkolesnik Can you re-review please?

**Review by skitt — APPROVED** [review]

File: src/content/development/building-testing/ci-maintenance/_index.en.md
Comment (dfarrell07) [inline src/content/development/building-testing/ci-maintenance/_index.en.md]:
```suggestion
Full Kubernetes Support | All non-latest supported versions | Workflow to run all CI for all of the currently-supported, non-latest Kubernetes versions. Run periodically, on releases, or manually by adding a label.
```

This does exist now, everything is done for this task except waiting on the re-review.

https://github.com/submariner-io/submariner-website/issues/624#issuecomment-992808761

**Review by sridhargaddam — APPROVED** [review]

**Comment by skitt** [comment]
@mkolesnik please follow up on your change requests.

**Comment by nyechiel** [comment]
@mkolesnik last chance before we merge

**Pull request merged**
